File: SIN BDD/RESTFULL_DOTNET_CONUNI_GR06/05. CLIESCRITORIO/cliente_escritorio_rest_dotnet_conuni_gr06/src/Servicio/ServicioLongitud.cs
using Ec.Edu.Monster.Modelo;
using Ec.Edu.Monster.Utils;

namespace Ec.Edu.Monster.Servicio;

public class ServicioLongitud
{
    private readonly ClienteRest cliente = new();

    public Resultado MetrosAPies(double metros) => cliente.Convertir("longitud", "metrosAPies", metros);

    public Resultado KilometrosAMillas(double kilometros) => cliente.Convertir("longitud", "kilometrosAMillas", kilometros);

    public Resultado CentimetrosAPulgadas(double centimetros) => cliente.Convertir("longitud", "centimetrosAPulgadas", centimetros);

    public Resultado YardasAMetros(double yardas) => cliente.Convertir("longitud", "yardasAMetros", yardas);

    public Resultado MilimetrosAPulgadas(double milimetros) => cliente.Convertir("longitud", "milimetrosAPulgadas", milimetros);
}

File: SIN BDD/RESTFULL_JAVA_CONUNI_GR06/07. CLIMOVIL/cliente_movil_restful_java_conuni_gr06/app/src/main/java/ec/edu/monster/servicio/ServicioLongitud.java
package ec.edu.monster.servicio;

import ec.edu.monster.util.ClienteRest;
import org.json.JSONObject;

public class ServicioLongitud {

    public double metrosAPies(double metros) throws Exception {
        return invocar("/longitud/metros-a-pies", metros);
    }

    public double kilometrosAMillas(double kilometros) throws Exception {
        return invocar("/longitud/kilometros-a-millas", kilometros);
    }

    public double centimetrosAPulgadas(double centimetros) throws Exception {
        return invocar("/longitud/centimetros-a-pulgadas", centimetros);
    }

    public double yardasAMetros(double yardas) throws Exception {
        return invocar("/longitud/yardas-a-metros", yardas);
    }

    public double milimetrosAPulgadas(double milimetros) throws Exception {
        return invocar("/longitud/milimetros-a-pulgadas", milimetros);
    }

    private double invocar(String ruta, double valor) throws Exception {
        JSONObject respuesta = ClienteRest.get(ruta, valor);
        return respuesta.getDouble("valor");
    }
}
